client/client-multi/client-cpp/kaa/utils/ThreadPool.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef THREADPOOL_HPP_
+#define THREADPOOL_HPP_
+
+#include <atomic>
+#include <cstdlib>
+#include <vector>
+#include <thread>
+
+#include "kaa/KaaThread.hpp"
+#include "kaa/utils/KaaTimer.hpp"
+#include "kaa/utils/IThreadPool.hpp"
+
+namespace kaa {
+
+class ThreadPool : public IThreadPool {
+    friend class Worker;
+
+public:
+    ThreadPool() : ThreadPool(DEFAULT_WORKER_NUMBER) {}
```

Comment:
Do we really need to use delegating constructor? I'd not rely on compiler to reduce unnecessary function call.Why don't you use parameters by default?